pkg/envoy/rds.go

```diff
@@ -0,0 +1,114 @@
+package envoy
+
+import (
+	"fmt"
+
+	envoy_config_route_v3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*RouteDomainCheck)(nil)
+
+// RouteDomainCheck implements common.Runnable
+type RouteDomainCheck struct {
+	*v1.Pod
+	ConfigGetter
+	RouteName string
+}
+
+// Run implements common.Runnable
+func (l RouteDomainCheck) Run() error {
+	if l.ConfigGetter == nil {
+		log.Error().Msg("Incorrectly initialized ConfigGetter")
+		return ErrIncorrectlyInitializedConfigGetter
+	}
+
+	envoyConfig, err := l.ConfigGetter.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	if envoyConfig == nil {
+		return ErrEnvoyConfigEmpty
+	}
+
+	foundAnyRoutes := false
+	// If Pod was defined -- check if this pod is in the list of Envoy config routes.
+	foundIt := false
```

Comment:
could this be changed to something a bit more informative? foundPod or whatever best reflects the above comment?